x-pack/plugins/alerting/common/rule.ts

```diff
@@ -89,8 +90,23 @@ export interface RuleAction {
   };
 }
 
-export interface RuleAggregations {
-  alertExecutionStatus: { [status: string]: number };
+type IndexType = Record<string, unknown>;
+
+export interface AggregateOptions extends IndexType {
```

Comment:
not sure it is only related to Aggregation
```suggestion
export interface RuleOptionsBase extends IndexType {
```